### PR TITLE
Fixes #8991: wget doesn't always work

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/8991

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,10 +16,27 @@
...
-WGET := $(if $(PROXY), http_proxy=$(PROXY) ftp_proxy=$(PROXY)) /usr/bin/wget -q
+# Autodetect wget or curl or curl usage and proxy configuration
+PROXY_ENV = $(if $(PROXY), http_proxy=$(PROXY) ftp_proxy=$(PROXY))
+WGET = wget -q -O
+CURL = curl -s -o
+FETCH = fetch -q -o
+ifneq (,$(wildcard /usr/bin/curl))
+GET = $(PROXY_ENV) $(CURL)
+GET_OPT = --curl
+CPAN_OPT = --curl --no-wget
```
Comment (amousset) [inline Makefile]: Do we need the CPAN_OPT here?

Comment (peckpeck) [inline Makefile]: no

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
OK, merging this PR

File: Makefile [hunk 2]
```diff
@@ -16,10 +16,23 @@
 #
 #####################################################################################
 
-WGET := $(if $(PROXY), http_proxy=$(PROXY) ftp_proxy=$(PROXY)) /usr/bin/wget -q
+# Autodetect wget or curl or curl usage and proxy configuration
+PROXY_ENV = $(if $(PROXY), http_proxy=$(PROXY) ftp_proxy=$(PROXY))
+WGET = wget -q -O
+CURL = curl -s -o
```
Comment (bmwjanos) [inline Makefile]: you might want to have `curl` using `-L` ...

**Pull request merged**